plugins/pipelines/index.js

```diff
@@ -88,6 +89,54 @@ const pipelinesPlugin = {
                 !credentials.scope.includes('pipeline') || parseInt(id, 10) === parseInt(credentials.pipelineId, 10)
         );
 
+        /**
+         * Throws error if a credential does not have access to a pipeline
+         * If credential has access, resolves to true
+         * @method canAccessPipeline
+         * @param {Object}  credentials              Credential object from Hapi
+         * @param {String}  credentials.username     Username of the person logged in (or build ID)
+         * @param {String}  credentials.scmContext   Scm of the person logged in (or build ID)
+         * @param {Array}   credentials.scope        Scope of the credential (user, build, admin)
+         * @param {String}  pipelineId               Target pipeline ID
+         * @param {String}  permission               Required permission level
+         * @param {String}  app                      Server app object
+         * @return {Promise}
+         */
+        server.expose('canAccessPipeline', (credentials, pipelineId, permission, app) => {
+            const { username, scmContext, scope } = credentials;
+            const { userFactory, pipelineFactory } = app;
+
+            if (credentials.scope.includes('admin')) {
+                return Promise.resolve(true);
+            }
+
+            return pipelineFactory.get(pipelineId).then(pipeline => {
+                if (!pipeline) {
+                    throw boom.notFound(`Pipeline ${pipelineId} does not exist`);
+                }
+
+                if (!scope.includes('user') || !pipeline.private) {
```

Comment:
what if scope is pipeline?
do we need to add a check like this: https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/secrets/index.js#L61

Reply:
could be a different pipeline's token?